src/mainwindow.cpp

```diff
@@ -524,38 +524,60 @@ bool MainWindow::registerNativeMessagingAppManifest() {
+    manifestPaths << QString("%1/Chromium/NativeMessagingHosts/%2.json").arg(QStandardPaths::writableLocation(QStandardPaths::GenericDataLocation)).arg(name);
 #elif defined (Q_OS_LINUX)
-    QString manifestPath = QString("%1/.mozilla/native-messaging-hosts/%2.json").arg(QStandardPaths::writableLocation(QStandardPaths::HomeLocation)).arg(name);
+    manifestPaths << QString("%1/.mozilla/native-messaging-hosts/%2.json").arg(QStandardPaths::writableLocation(QStandardPaths::HomeLocation)).arg(name);
+    manifestPaths << QString("%1/.config/google-chrome/NativeMessagingHosts/%2.json").arg(QStandardPaths::writableLocation(QStandardPaths::HomeLocation)).arg(name);
+    manifestPaths << QString("%1/.config/chromium/NativeMessagingHosts/%2.json").arg(QStandardPaths::writableLocation(QStandardPaths::HomeLocation)).arg(name);
 #elif defined (Q_OS_WIN)
     // On Windows, we write the manifest to some safe directory, and then point to it from the Registry.
-    QString manifestPath = QString("%1/%2.json").arg(QStandardPaths::writableLocation(QStandardPaths::AppConfigLocation)).arg(name);
-    QSettings registry(QSettings::NativeFormat, QSettings::UserScope, "Mozilla", "NativeMessagingHosts");
-    registry.setValue(QString("%1/Default").arg(name), manifestPath);
+    manifestPaths << QString("%1/%2.json").arg(QStandardPaths::writableLocation(QStandardPaths::AppConfigLocation)).arg(name);
 #else
     return false;
 #endif
 
-    QFileInfo manifestInfo(manifestPath);
+    for (QString const &manifestPath : manifestPaths) {
+        QFileInfo manifestInfo(manifestPath);
 
-    if (!manifestInfo.dir().exists()) {
-        if (!QDir().mkpath(manifestInfo.absolutePath())) {
-            qDebug() << "Cannot create directory:" << manifestInfo.absolutePath();
-            return false;
+        if (!manifestInfo.dir().exists()) {
+            if (!QDir().mkpath(manifestInfo.absolutePath())) {
+                qDebug() << "Cannot create directory:" << manifestInfo.absolutePath();
+                return false;
+            }
         }
+
+        QFile manifestFile(manifestPath);
+        manifestFile.open(QFile::WriteOnly);
+        manifestFile.write(manifest.toJson());
     }
 
-    QFile manifestFile(manifestPath);
-    manifestFile.open(QFile::WriteOnly);
-    manifestFile.write(manifest.toJson());
+#if defined (Q_OS_WIN)
+    QStringList registryKeys;
+    registryKeys << QString("HKEY_CURRENT_USER\\Software\\Mozilla\\NativeMessagingHosts\\%1").arg(name);
+    registryKeys << QString("HKEY_CURRENT_USER\\Software\\Google\\Chrome\\NativeMessagingHosts\\%1").arg(name);
+    registryKeys << QString("HKEY_CURRENT_USER\\Software\\Chromium\\NativeMessagingHosts\\%1").arg(name);
+
+    for (QString const &key : registryKeys)
+        QSettings(key, QSettings::NativeFormat).setValue("Default", manifestPath);
```

Comment:
You meant for this to be inside the for loop above?

Reply:
For Windows I only write the manifest file itself only once and then register all the different registry keys to point to the same file.

But you're right that it is weird to see `manifestPath` outside of the for-loop. Its probably safer to just move all of this inside the for-loop since `manifestPaths` is `size() == 1` anyway in Win.

Reply:
Well, this is why it doesn't compile on windows, but yeah I'd say it should be in the for loop for clarity's sake.

Reply:
Oh. Ooops 😬